x-pack/plugins/security_solution/public/app/upgrade.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import semver from 'semver';
+import { useEffect } from 'react';
+import { HttpFetchOptions, HttpStart } from 'src/core/public';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import {
+  GetInfoResponse,
+  epmRouteService,
+  PackageInfo,
+  InstallPackageRequest,
+  InstallationStatus,
+  appRoutesService,
+  CheckPermissionsResponse,
+  InstallPackageResponse,
+} from '../../../ingest_manager/common';
+import { sendGetEndpointSecurityPackage } from '../management/pages/policy/store/policy_list/services/ingest';
+import { StartServices } from '../types';
+import { useIngestEnabledCheck } from '../common/hooks/endpoint/ingest_enabled';
+
+/**
+ * Retrieves the info for a package from the ingest manager
+ *
+ * @param http an http client for sending the request
+ * @param packageKey a string in the form of <package name>-<package version>
+ * @param options an object containing options for the request
+ */
+const sendGetPackageInfo = async (
```

Comment:
@oatkiller @paul-tavares where should these functions live? The management and policy pages have some similar functions and are within a `services` directory under those pages https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/management/pages/policy/store/policy_list/services/ingest.ts#L137

Reply:
I would defer to @XavierM or @spong for this. I don't know about the structure of this part of the app.